domain: add pinnings support to GetVCpus()

Leonid99 opened the pull request.

hinesmr commented:
LGTM.  @rgbkrk comments?

rgbkrk commented:
Thanks for rebasing, I almost commented on an old version of this since it was sitting in an idle tab.

rgbkrk commented on domain.go:
```diff
@@ -676,28 +676,83 @@ func (d *VirDomain) MemoryStats(nrStats uint32, flags uint32) ([]VirDomainMemory
+	return out
+}
+
+func (d *VirDomain) GetVcpus(conn *VirConnection, maxInfo int32) ([]VirVcpuInfo, [][]uint32, error) {
```
This is changing the interface to a function that previously only took `maxInfo` - should we write a new function and aim to deprecate the old one?

rgbkrk replied:
@hinesmr ^^ my only concern here

rgbkrk replied:
Since this used to be `func (d *VirDomain) GetVcpus(maxInfo int32) ([]VirVcpuInfo, error) {` (see https://github.com/rgbkrk/libvirt-go/pull/112/files/8f066a0e9dfb9fe811173e71f99518b8070b151a#diff-f86efe7096c9bab5707c9d34a313c51eL679)

Leonid99 replied:
@rgbkrk @hinesmr  I'm feeling that an extra argument is less worrying, since you always have a connection when calling this function. I'm more worried about an extra parameter returned, I could fold the CPUs mask into `VirVcpuInfo` to avoid changing callers code

rgbkrk commented on nodeinfo.go:
```diff
@@ -48,3 +48,8 @@ func (ni *VirNodeInfo) GetCores() uint32 {
+// libvirt.h: VIR_NODEINFO_MAXCPUS
+func (ni *VirNodeInfo) GetMaxCPUs() uint32 {
+	return ni.GetNodes() * ni.GetSockets() * ni.GetCores() * ni.GetThreads()
+}
```
👍

hinesmr commented:
@rgbkrk @Leonid99 is going to re-work this and break-out the individual libvirt functions into separate exports the same way it's done on the virsh client-side. The original will GetVcpus will stay the same.

Leonid99 commented:
@hinesmr @rgbkrk I reworked this code quite a bit and also added support for 2 more functions. I'm closing this PR and creating a new one. Sorry for the noise.
